src/apps/config/leader.lua

```diff
@@ -92,8 +93,24 @@ function Leader:start_worker()
    return worker.start("follower", self.worker_start_code)
 end
 
-function Leader:stop_worker(worker)
-   S.kill(worker.pid, 15)
+function Leader:stop_worker(id)
+   -- Tell the worker to terminate
+   local stop_actions = {{'shutdown', {}}, {'commit', {}}}
+   self:enqueue_config_actions_for_follower(id, stop_actions)
```

Comment:
Can we manage the "check_stale" flag here?  I.e. set `self.followers[id].shutting_down = true` here, then below....